### DISCUSSION TITLE
Programmatically click button

### DISCUSSION THREAD

**Discussion by SergeyYakubov** [Kitware/trame#513] [Q&A]
Hi, that might be very simple, but I cannot find an answer myself.

How can I trigger a button click programmatically in Trame? E.g., a user clicks one button, and in the on_click function, I trigger the clicking of another button.

More specifically, I want a user to click a button, then I execute some function to find and construct a URL, set a href for a second (hidden) button, and click it. The result should be a new window with the specified url opened in the browser.

**Comment by jourdain** [comment]
They are several ways to do that... 
One simple way would be to do the same thing we do with [downloading a file](https://kitware.github.io/trame/examples/core/files.html). 
Otherwise, you can update a reactive variable and then listen to it via a component to execute some JS code when it change...

But based on what you are describing, only the first one will work due to security limitation in web.

**Reply by SergeyYakubov** [reply]
Could you give an example how to do it "similar to download"? Is there a utils.open function?

**Reply by jourdain** [reply]
No utils.open, but you can register [your own function](https://github.com/Kitware/trame/tree/master/examples/09_advanced/custom_js).
